sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/SetFns.java

```diff
@@ -0,0 +1,261 @@
+    return new SetImpl<>(rightCollection, exceptFn);
+  }
+
+  /**
+   * Returns a new {@code SetFns.SetImpl<T>} transform that compute the difference all (exceptAll)
+   * with provided {@code PCollection<T>}.
+   *
+   * <p>The argument should not be modified after this is called.
+   *
+   * <p>The elements of the output {@link PCollection<T>} which will follow EXCEPT_ALL Semantics as
+   * follows: Given there are m elements on pipeline which is constructed {@link PCollection<T>}
+   * (left) and n elements on in provided {@link PCollection<T>} (right): - it will output m
+   * elements of left for all elements which are present in left but not in right. - it will output
+   * MAX(m - n, 0) elements of left for all elements which are present in both left and right.
+   *
+   * <pre>{@code
+   * Pipeline p = ...;
+   *
+   * PCollection<String> left = p.apply(Create.of("1", "2", "3", "4", "5"));
+   * PCollection<String> right = p.apply(Create.of("1", "3", "4", "6"));
+   *
+   * PCollection<String> results =
+   *     left.apply(SetFns.exceptAll(right));
+   * }</pre>
+   */
+  public static <T> SetImpl<T> exceptAll(PCollection<T> rightCollection) {
+    checkNotNull(rightCollection, "rightCollection argument is null");
+    SerializableBiFunction<Long, Long, Long> exceptFn =
+        (numberOfElementsinLeft, numberOfElementsinRight) -> {
+          if (numberOfElementsinLeft > 0 && numberOfElementsinRight == 0) {
+            return numberOfElementsinLeft;
+          } else if (numberOfElementsinLeft > 0 && numberOfElementsinRight > 0) {
+            return Math.max(numberOfElementsinLeft - numberOfElementsinRight, 0L);
+          }
+          return 0L;
+        };
+    return new SetImpl<>(rightCollection, exceptFn);
+  }
+
+  /**
+   * Returns a new {@code SetFns.SetImpl<T>} transform that compute the union with provided {@code
+   * PCollection<T>}.
+   *
+   * <p>The argument should not be modified after this is called.
+   *
+   * <p>The elements of the output {@link PCollection<T>} will all distinct elements that present in
+   * pipeline is constructed {@link PCollection<T>} and {@link PCollection<T>}.
+   *
+   * <pre>{@code
+   * Pipeline p = ...;
+   *
+   * PCollection<String> left = p.apply(Create.of("1", "2", "3", "4", "5"));
+   * PCollection<String> right = p.apply(Create.of("1", "3", "4", "6"));
+   *
+   * PCollection<String> results =
+   *     left.apply(SetFns.union(right));
+   * }</pre>
+   */
+  public static <T> SetImpl<T> union(PCollection<T> rightCollection) {
+    checkNotNull(rightCollection, "rightCollection argument is null");
+    SerializableBiFunction<Long, Long, Long> unionFn = (numberOfElementsinLeft, numberOfElementsinRight) -> 1L;
+    return new SetImpl<>(rightCollection, unionFn);
+  }
+
+  /**
+   * Returns a new {@code SetFns.SetImpl<T>} transform that compute the unionAll with provided
```

Comment:
Isn't this just `Flatten.pcollections`? (Also, isn't that more efficient?)

Reply:
Thanks very valuable. I have fixed this.